### ISSUE TITLE
(3) Implement restriction and display for users on Campus IP

### ISSUE THREAD

**Issue by rmjaffe** [UCSCLibrary/dams_project_mgmt#246]
### User Story 

As a User on Campus IP addresses (determined in https://github.com/UCSCLibrary/dams_project_mgmt/issues/241) I can access (view and play) CMF records 

### Descriptive Summary

Limit access to certain content (i.e. collections) based on IP range.

### Background 

Goal of Tier 1 implementation [https://docs.google.com/document/d/1Y-jevhML30_m8aHEd8joPib5T3r4FL8HkZ2cB1QTg-c/edit](url).

### Acceptance Criteria

Users on Campus IP Steps:
* [ ] User navigates to CMF record
* [ ] User immediately access (ie. view and play) object

**Comment by rmjaffe** [comment]
@NedHenry This ticket leaves a lot to be desire and I'm guess may end up being multiple tickets.  Please consider it a placeholder.

**Comment by NedHenry** [comment]
I have implemented these changes and deployed the code updates to the production server. 
Currently, all but one of the Cabrillo Music Festival works are private, and are only visible to administrators either on or off campus. To do a brief test of the functionality, I made one work from this festival visible to users on campus. 
[Here is the link to that work](https://digitalcollections.library.ucsc.edu/concern/works/ws859h956), and [here is a link to the collection page](https://digitalcollections-staging.library.ucsc.edu/collections/9s161867). 
When you are logged in as an admin, you should be able to see the work and stream the audio whether you are on campus or not.
When you are not logged in and are on campus, you should be able to see the work and stream the audio.
When you are not logged in and are off campus, you should be able to see the work and its metadata but not stream the audio. 
The other works in this collection (including the parent work and sibling works of this one) are still private, and viewing them will prompt you to log in. 
When the team is happy with this update (and with the Cabrillo collection), let me know and I can easily publish the rest of the Cabrillo Music Festival data for on-campus streaming.

**Comment by rmjaffe** [comment]
Awesome! Just very quickly checked these out:

When I am logged in as an admin, I could see and stream the audio.
When I am not logged in and are on campus (via VPN), I could see and stream the audio.
When I am not logged in and are off campus, I could see and NOT stream the audio.

And yes, when click on the related works that are still private, I was prompted to log in.

**Comment by chesleyperry** [comment]
Logged in as an admin, I could see full metadata record and stream the audio.  Also could stream audio for other files in the record.

Not logged in as admin and on campus (via VPN): I could see full metadata record and stream the audio for part 2 only.  Was promoted to login to listen to other parts of the concert.

Not logged in as admin and off campus (no VPN), I could see the metadata record but NOT stream the audio.

Collection page looks good for now - Ryan is working on moving it into the new mockup within the next sprint.  I think it's fine to publish as is for QA.

**Comment by tmariamora** [comment]
So when I first saw this message in slack I had the experience that RJ and SCP report above. However, I just tested again in order to document here and I have full access to search and stream all recordings while not logged into admin or the VPN. I've tested in three separate internet browsers to confirm this wasn't some weird cache issue and am granted access. 

Also just tried on my phone and the behavior was as expected: prompted to login... so maybe this is some very weird cache issue?

Sorry.
